src/plugin.rs

```diff
@@ -20,7 +20,21 @@ pub fn run_plugins(
             .and_then(|cmd| cmd.as_str().map(|s| s.to_owned()))
             .unwrap_or_else(|| format!("yarner-{}", name));
 
-        let args = shlex::split(&command).ok_or(format!("Invalid plugin command {}", command))?;
+        let arguments: Vec<&str> = match config.get("arguments") {
+            None => vec![],
+            Some(v) => v
+                .as_array()
+                .map(|arr| arr.iter().map(|l| l.as_str().unwrap_or_default()))
+                .ok_or("Can't parse array of plugin arguments")?
+                .collect(),
+        };
+
+        let command_string = format!(
+            "{}{}{}",
+            command,
```

Comment:
Couldn't you just `iter::once(command).chain(arguments).join(" ")`?

Reply:
Didn't know of this "syntax"...

Although it is a little more complicated...

```rust
let command_string = std::iter::once(&command.as_str())
    .chain(&arguments)
    .join(" ", "");
```

Reply:
It is a matter of preference I guess. One might even drop printing the arguments entirely as the printed command might also be misleading without quoting the individual arguments (spaces and such).

Reply:
I think the chain version is better than the previous one, particularly due to the condition inside `format!`. Your second point is definitely worth considering.

Reply:
Simplified plugin log messages and removed this.